code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -565,20 +565,17 @@
 	. = ..()
 
 /datum/dynamic_ruleset/midround/from_ghosts/nightmare/generate_ruleset_body(mob/applicant)
-	var/datum/mind/player_mind = new /datum/mind(applicant.key)
-	player_mind.active = TRUE
-
-	var/mob/living/carbon/human/S = new (pick(spawn_locs))
-	player_mind.transfer_to(S)
-	player_mind.set_assigned_role(SSjob.GetJobType(/datum/job/nightmare))
-	player_mind.special_role = ROLE_NIGHTMARE
-	player_mind.add_antag_datum(/datum/antagonist/nightmare)
-	S.set_species(/datum/species/shadow/nightmare)
-
-	playsound(S, 'sound/magic/ethereal_exit.ogg', 50, TRUE, -1)
-	message_admins("[ADMIN_LOOKUPFLW(S)] has been made into a Nightmare by the midround ruleset.")
-	log_game("DYNAMIC: [key_name(S)] was spawned as a Nightmare by the midround ruleset.")
-	return S
+	var/mob/living/carbon/human/nightmare = new (pick(spawn_locs))
+	nightmare.key = applicant.key
```

Comment:
Problem with these is Login will not be called if the client is not present, so these can still runtime given new will yield and client dels can happen then. Ideally we'd have a set_key() wrapper to handle these.

Reply:
This should never be a problem here because applicants without a client already get removed from the list at the end of `pollGhostCandidates`, and no proc gets delayed/sleeped after that.